bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/JavaAspectsClasspathInfo.java

```diff
@@ -363,6 +367,10 @@ public Collection<ClasspathEntry> compute() throws CoreException {
 
         // Collect jars referenced by runtime deps
         for (TargetKey targetKey : runtimeDeps) {
+            if ((availableDependencies.size() > 0)
+                    && !availableDependencies.contains(new BazelLabel(targetKey.getLabel().toString()))) {
+                continue;
```

Comment:
* I am not particular happy with the name. It does not convey the intent/purpose clear. Perhaps `runtimeDependencyIncludes` is better?
* Also, consider moving this into a method for readability (eg., `shouldExcludeRuntimeDependency`).
* Can you add comment explaining the motivation, i.e. is it memory impact, performance, etc.?